src/shared/components/common/loading-skeleton.tsx

```diff
@@ -0,0 +1,155 @@
+import { Component } from "inferno";
+
+interface LoadingSkeletonProps {
+  itemCount?: number;
+}
+
+interface LoadingSkeletonLineProps {
+  size: 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12;
+}
+
+class LoadingSkeletonLine extends Component<LoadingSkeletonLineProps, any> {
+  render() {
+    const className = "placeholder placeholder-lg col-" + this.props.size;
+    return (
+      <p className="placeholder-glow" style={{ margin: "0" }}>
+        <span className={className} style={{ height: "1.3rem" }}></span>
+      </p>
+    );
+  }
+}
+
+export class PostsLoadingSkeleton extends Component<LoadingSkeletonProps, any> {
+  render() {
+    return Array.from({ length: this.props.itemCount ?? 10 }, (_, index) => (
+      <PostsLoadingSkeletonItem key={index} />
+    ));
+  }
+}
+
+class PostThumbnailLoadingSkeleton extends Component<any, any> {
+  render() {
+    return (
+      <div className="thumbnail rounded d-flex justify-content-center placeholder-glow">
+        <span
+          className="placeholder placeholder-lg"
+          style={{ height: "100%", width: "100%", "border-radius": "0.5rem" }}
+        ></span>
+      </div>
+    );
+  }
+}
+
+class PostsLoadingSkeletonItem extends Component<any, any> {
+  render() {
+    return (
+      <>
+        {/* mobile */}
+        <div className="d-block d-sm-none">
+          <div className="col flex-grow-1 mt-2">
+            <div className="row">
+              <div className="col flex-grow-1">
+                <LoadingSkeletonLine size={12} />
+                <LoadingSkeletonLine size={8} />
+                <LoadingSkeletonLine size={6} />
+              </div>
+              <div className="col flex-grow-0">
+                <PostThumbnailLoadingSkeleton />
+              </div>
+            </div>
+          </div>
+        </div>
+        {/* desktop */}
+        <div className="d-none d-sm-block my-3">
+          <div className="col flex-grow-1">
+            <div className="row">
+              <div className="col flex-grow-0">
+                <PostThumbnailLoadingSkeleton />
+              </div>
+              <div className="col flex-grow-1">
+                <LoadingSkeletonLine size={12} />
+                <LoadingSkeletonLine size={8} />
+                <LoadingSkeletonLine size={4} />
+              </div>
+            </div>
+          </div>
+        </div>
+      </>
+    );
+  }
```

Comment:
I don't think the duplication is required. You should be able to get away with:
```suggestion
  render() {
    return (
        <div className="my-3">
          <div className="col flex-grow-1">
            <div className="row">
              <div className="col flex-grow-0 order-last order-sm-first">
                <PostThumbnailLoadingSkeleton />
              </div>
              <div className="col flex-grow-1">
                <LoadingSkeletonLine size={12} />
                <LoadingSkeletonLine size={8} />
                <LoadingSkeletonLine size={4} />
              </div>
            </div>
          </div>
        </div>
    );
  }
```

Reply:
Indeed. Thank you very much for the suggestion